src/base/bittorrent/filterparserthread.cpp

```diff
@@ -37,75 +42,133 @@
+        if constexpr (isDat) // DAT format
+        {
+            // Each line should follow this format:
+            // 001.009.096.105 - 001.009.096.105 , 000 , Some organization
+            // The 2nd entry is access level and if >=127 the IP range isn't blocked.
+            if (const auto firstComma = line.find(',');
+                firstComma != line.npos)
             {
-                if (*str == '.')
-                {
-                    const long int extractedNum = strtol(octetStart, &endptr, 10);
-                    if ((extractedNum >= 0L) && (extractedNum <= 255L))
-                        m_buf[octetIndex++] = static_cast<unsigned char>(extractedNum);
-                    else
-                        return false;
-
-                    if (endptr != str)
-                        return false;
-                    if (octetIndex == 4)
-                        return true;
-
-                    octetStart = str + 1;
-                }
-            }
+                entry.range = line.substr(0, firstComma);
+
+                auto posIt  = std::find_if_not(line.cbegin() + firstComma + 1, line.cend(),
+                                              [](unsigned char c){ return std::isspace(c); } );
+                int pos = 0;
+                if (posIt != line.cend())
+                    pos = posIt - line.cbegin();
 
-            if (str != octetStart)
+                // from_chars will stop when it encounters non-numeric chars
+                // we don't care for any other conversion failure
+                std::from_chars(line.data() + pos,
+                                line.data() + line.size(),
+                                entry.accessLevel);
+            }
+            else
             {
-                const long int extractedNum = strtol(octetStart, &endptr, 10);
-                if ((extractedNum >= 0L) && (extractedNum <= 255L))
-                    m_buf[octetIndex] = static_cast<unsigned char>(strtol(octetStart, &endptr, 10));
-                else
-                    return false;
-
-                if ((endptr == str) && (octetIndex == 3))
-                    return true;
+                // Apparently only range is mandatory
+                entry.range = line;
             }
+        }
+        else // P2P format
+        {
+            // Each line should follow this format:
+            // Some organization:1.0.0.0-1.255.255.255
+            // The "Some organization" part might contain a ':' char itself so we find the last occurrence
+            // Apparently this format wasn't designed for IPv6, so no effort will be made
+            // to parse possible IPv6 literals wrapped in []. These characters can appear in
+            // the "some organization" part too. It isn't worth implementing such complex parsing
+            // for a format that doesn't support IPv6 and there aren't any lists with IPv6 in them.
+            if (const auto colon = line.rfind(':');
```

Comment:
It unclear what type the function returns, I wouldn't use `auto` here.

Reply:
>It unclear what type the function returns,

It's just one of standard iterator-related methods like (c/r)begin/(c/r)end etc.

But seems code in this line does nothing... There is empty body in `if`. Do I miss something?

Reply:
> There is empty body in if. Do I miss something?

uhm this is an if with an init statement. This line is the initstatement and the next one the condition. The body follows. (Look at the open/close parentheses).

Reply:
>uhm this is an if with an init statement. This line is the initstatement and the next one the condition. The body follows. 

Ooops... I didn't notice that it's splitted into several lines.